opm/autodiff/NewtonSolver.hpp

```diff
@@ -66,7 +66,7 @@ namespace Opm {
         /// \param[in]      param   parameters controlling nonlinear Newton process
         /// \param[in, out] model   physical simulation model
         explicit NewtonSolver(const SolverParameters& param,
-                              PhysicalModel& model);
+                              std::shared_ptr<PhysicalModel> model);
```

Comment:
The correctness argument is that you should use a reference unless you will actually store the pointer and thereby participate in lifecycle mangement. Since you actually store it, using it as a function argument is ok actually, so my question should have been: why have you changed the *data member* to be a shared_ptr? I do not see the need upfront, did you change something in the way the classes are used that makes you require this?